jlm/hls/backend/rvsdg2rhls/mem-conv.cpp

```diff
@@ -710,6 +717,11 @@ jlm::hls::MemoryConverter(jlm::llvm::RvsdgModule & rm)
   // once.
   //
   remove_unused_state(root);
+
+  // Need to get the lambda from the root since remote_unused_state replaces the lambda
+  JLM_ASSERT(root->nnodes() == 1);
+  newLambda = jlm::util::AssertedCast<jlm::llvm::lambda::node>(root->nodes.first());
```

Comment:
Could you use `*root->Nodes().begin()` instead of `root->nodes.first()`. The plan is to privatize the nodes attribute in the future.

Reply:
My C++ skills make using the API frustrating. Changing to what you suggest gives the following error:
./jlm/util/common.hpp:51:1: note: candidate template ignored: could not match 'From *' against 'jlm::rvsdg::node'
   51 | AssertedCast(From * value)

Reply:
Try: `root->Nodes().begin().ptr()`

Reply:
Works